hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDWriteClient.java

```diff
@@ -438,33 +424,48 @@ private void writeTableMetadataForTableServices(HoodieTable<T, JavaRDD<HoodieRec
     UpgradeDowngrade upgradeDowngrade = new UpgradeDowngrade(
         metaClient, config, context, SparkUpgradeDowngradeHelper.getInstance());
     if (upgradeDowngrade.needsUpgradeOrDowngrade(HoodieTableVersion.current())) {
```

Comment:
lets move the check also into the lock? otherwise two writers would think they both need to upgrade and they just wait on the lock is all.

Reply:
fixed.